### PR TITLE
Cargo dep fix

### PR DESCRIPTION
**Pull request by melynx**
Updated the project to use v1.1.1 of rust-sgx-sdk.
Fixed up the Cargo dependencies.
Currently using patch to ensure that all remote dependencies uses the local copy.

NOTE: Things will break if rust-sgx-sdk move up a version. Patching doesn't work for an older version. protobuf-rust is also fixed at v2.8 and might break in the future.

### CODE REVIEW

**Comment by melynx** [comment]
I've made the appropriate changes in commit b4f11d3.

File: enclave/Cargo.toml [hunk 1]
```diff
@@ -14,64 +14,58 @@ default = []
...
 [dependencies]
 # unmodified crates
 codec             = { package = "parity-scale-codec", version = "1.0.0", default-features = false, features = ["derive"] }
 primitive-types   = { version = "0.6", default-features = false, features = ["codec"] }
 
 # mesalock-modified crates
-log               = { tag = "sgx_1.1.0", git = "https://github.com/mesalock-linux/log-sgx" }
-env_logger        = { tag = "sgx_1.1.0", git = "https://github.com/mesalock-linux/env_logger-sgx" }
+# NOTE: Have to ensure all the git repos have the same Cargo version
+log               = { git = "https://github.com/mesalock-linux/log-sgx" }
```
Comment (Gnnng) [inline enclave/Cargo.toml:34]: missed `tag` field

File: enclave/Cargo.toml [hunk 2]
```diff
@@ -14,64 +14,58 @@ default = []
...
+env_logger        = { git = "https://github.com/mesalock-linux/env_logger-sgx" }
```
Comment (Gnnng) [inline enclave/Cargo.toml:35]: `tag = "sgx_1.1.0"` field is missing

File: enclave/Cargo.toml [hunk 3]
```diff
@@ -14,64 +14,58 @@ default = []
...
 serde        	  = { git = "https://github.com/mesalock-linux/serde-sgx" }
-serde_json        = { tag = "sgx_1.1.0", git = "https://github.com/mesalock-linux/serde-json-sgx"}
+serde_json        = { git = "https://github.com/mesalock-linux/serde-json-sgx"}
```
Comment (Gnnng) [inline enclave/Cargo.toml:37]:
```suggestion
serde_json        = { tag = "sgx_1.1.0", git = "https://github.com/mesalock-linux/serde-json-sgx"}
```

**Pull request merged**